finite-state-machine/src/main/java/com/iluwatar/finite/state/machine/states/CorrectFirstLetterState.java

```diff
@@ -0,0 +1,87 @@
+package com.iluwatar.finite.state.machine.states;
+
+import com.iluwatar.finite.state.machine.AutomatonInterfaceI;
+import com.iluwatar.finite.state.machine.DataModel;
+import com.iluwatar.finite.state.machine.Event;
+import com.iluwatar.finite.state.machine.EventSink;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * State when first letter was correct.
+ */
+@Slf4j
+public class CorrectFirstLetterState implements AutomatonInterfaceI {
+
+  /**
+   * Event sink for sending state changes.
+   * */
+  private final EventSink eventSink;
+
+  /**
+   * Event sink for current inputs value.
+   * */
+  private final DataModel model;
+
+
+  /**
+   * Constructor.
+   *
+   * @param newEventSink -> evenSink value.
+   * @param newModel -> model value.
+   * */
+  public CorrectFirstLetterState(
+      final EventSink newEventSink,
+      final DataModel newModel
+  ) {
+    this.eventSink = newEventSink;
+    this.model = newModel;
+  }
+
+  /**
+   * Restarts the FSM.
+   * */
+  @Override
+  public void startNewQuery() {
+    LOGGER.info("Input cache cleared");
+    eventSink.castEvent(Event.CLEAR);
+    model.clear();
+  }
+
+  /**
+   * Input new character into FSM.
+   *
+   * @param character -> value of inputted character.
+   * */
+  @Override
+  public void inputCharacter(final Character character) {
+    // check if character is a lowercase letter
+    if (Character.isLowerCase(character)) {
+      eventSink.castEvent(Event.CORRECT);
+    } else {
+      eventSink.castEvent(Event.INCORRECT);
+    }
```

Comment:
Make it shorter by using ternary operator

Reply:
Sure thing, I even though about them briefly, but wasn't sure if the syntax isn't to esoteric